bench/workgen/workgen.cxx

```diff
@@ -248,6 +281,231 @@ WorkloadRunner::start_table_idle_cycle(WT_CONNECTION *conn)
+            }
+
+            // The dynamic table data structures are protected by a mutex.
+            {
+                const std::lock_guard<std::mutex> lock(*icontext->_dyn_mutex);
+
+                // Add the table into the list of dynamic set.
+                tint_t tint = icontext->_dyn_tint_last;
+                icontext->_dyn_tint[uri] = tint;
+                icontext->_dyn_table_names[tint] = uri;
+                icontext->_dyn_table_runtime[tint] = TableRuntime();
+                ++icontext->_dyn_tint_last;
+                VERBOSE(*_workload, "Created table and added to the dynamic set: " << uri);
+            }
+        }
+
+        sleep(_workload->options.dynamic_create_period);
+    }
+
+    return 0;
+}
+
+/*
+ * This function will drop "dynamic_drop_count" tables every "dynamic_drop_period" seconds. It will
+ * also monitor the database size and stop dropping tables once the size goes below
+ * "dynamic_drop_target". It will restart table drops if the database size crosses
+ * "dynamic_drop_trigger".
+ */
+int
+WorkloadRunner::start_dynamic_drop(WT_CONNECTION *conn)
+{
+    WT_SESSION *session;
+    if (conn->open_session(conn, nullptr, nullptr, &session) != 0) {
+        THROW("Error opening a session.");
+    }
+
+    WT_DECL_RET;
+    ContextInternal *icontext = _workload->_context->_internal;
+    std::vector<std::string> tables_delete; // Track tables that are to be deleted
+    uint32_t db_size = get_dir_size_mb(_wt_home);
+
+    // Initially we start dropping tables if the database size is more than the drop trigger.
+    bool dropping = db_size > _workload->options.dynamic_drop_trigger;
+
+    while (!stopping) {
+        /*
+         * If we have been dropping, continue until we reach the drop target size. If we have not
+         * been dropping tables, begin to do so if the database size crosses the drop trigger.
+         */
+        db_size = get_dir_size_mb(_wt_home);
+        if (dropping) {
+            dropping = db_size > _workload->options.dynamic_drop_target;
+            if (!dropping) {
+                VERBOSE(*_workload,
+                  "Stopped dropping new tables. db_size now "
+                    << db_size << " MB has reached the drop target of "
+                    << _workload->options.dynamic_drop_target << " MB.");
+            }
+        } else {
+            dropping = db_size > _workload->options.dynamic_drop_trigger;
+            if (dropping) {
+                VERBOSE(*_workload,
+                  "Started dropping new tables. db_size now "
+                    << db_size << " MB has reached the drop trigger of "
+                    << _workload->options.dynamic_drop_trigger << " MB.");
+            }
+        }
+        if (!dropping) {
+            sleep(_workload->options.dynamic_drop_period);
+            continue;
+        }
+
+        for (int drops = 0; drops < _workload->options.dynamic_drop_count; ++drops) {
```

Comment:
We should check for `stopping` here too.

Reply:
Look at the comment before, this is intentional. A restructure in follow-on work is planned.